specs/fault-proof/stage-one/anchor-state-registry.md

```diff
@@ -0,0 +1,224 @@
+
+When asked if a game is a Proper Game, the `AnchorStateRegistry` must serve a response that is
+identical to the response that would be given by a perfect oracle for this query.
+
+#### Impact
+
+**Severity: High**
+
+If this invariant is broken, the anchor state could be set to an incorrect value, which would cause
+future Dispute Game instances to use an incorrect starting state. This would lead games to resolve
+incorrectly.
+
+#### Dependencies
+
+- [aASR-001](#aasr-001-dispute-game-contracts-properly-report-important-properties)
+- [aASR-002](#aasr-002-dispute-game-factory-properly-reports-its-created-games)
+- [aASR-003](#aasr-003-optimismportal-properly-reports-respected-game-type-blacklist-and-retirement-time)
+
+## Function Specification
+
+### isGameRegistered
+
+Determines if a game is a Registered Game.
+
+- MUST return `true` if and only if the game was created by the system's `DisputeGameFactory` contract.
+
+### isGameRespected
+
+Determines if a game is a Respected Game.
+
+- MUST return `true` if and only if the game's game type is the respected game type defined by the
+  `OptimismPortal` contract as per a call to `OptimismPortal.respectedGameType()`.
+
+### isGameBlacklisted
+
+Determines if a game is a Blacklisted Game.
+
+- MUST return `true` if and only if the game's address is marked as blacklisted inside of the
+  `OptimismPortal` contract as per a call to `OptimismPortal.disputeGameBlacklist(game)`.
+
+### isGameRetired
+
+Determines if a game is a Retired Game.
+
+- MUST return `true` if and only if the game was created before the retirement timestamp defined by
+  the `OptimismPortal` contract as per a call to `OptimismPortal.respectedGameTypeUpdatedAt()`.
+  Check should be a strict comparison that the creation is less than the retirement timestamp.
+
+### isGameProper
+
+Determines if a game is a Proper Game.
+
+- MUST return `true` if and only if `isGameRegistered(game)` and `isGameRespected(game)` are both
+  `true` and `isGameBlacklisted(game)` and `isGameRetired(game)` are both `false`.
+
+### getAnchorRoot
+
+Retrieves the current anchor root.
+
+- MUST return the root hash and L2 block height of the current anchor state.
+
+### anchors
+
+Legacy function. Accepts a game type as a parameter but does not use it.
+
+- MUST return the current value of `getAnchorRoot()`.
+
+### tryUpdateAnchorState
+
+Allows a `DisputeGame` contract to attempt to set its own result as the anchor state. The
+`AnchorStateRegistry` contract previously stored the anchor state separately for each game type.
+The updated version of this function allows any game type to update the anchor state as long as it
+is the currently respected game type.
+
+- MUST not revert if the caller is a Registered Game and sufficient gas is provided.
+- MUST not update the anchor state if the game is not a Proper Game for any reason.
+- MUST not update the anchor state if the game resolved in favor of the Challenger.
```

Comment:
This isn't strict enough.  It must not update if the game is in any state other than defender wins.  You must not update the anchor state with an In Progress game.

Reply:
Good point, updating